06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js

```diff
@@ -0,0 +1,91 @@
+
+/**
+ * InvertedIndex class with constructor
+ * @class
+ */
+class InvertedIndex {
+
+  /**
+   * Creates an instance of InvertedIndex.
+   * @memberOf InvertedIndex
+   */
+  constructor() {
+    this.indexedFiles = {};
+    this.searchResults = {};
+  }
+
+  /** creates index and update the indexed files
+   * @param {object} uploadedFiles - object containing uploaded books
+   * @param {Sting} file - file name
+   * @return  {object}  - this.indexedFiles
+   */
+  updateIndexedFilesRecords(uploadedFiles, file) {
+    this.selectedBook = uploadedFiles[file].content;
+    this.validateContent = helpers.validFileContent(this.selectedBook);
+    if (this.validateContent) {
```

Comment:
Why do you have a bunch of instance variable in this function when they're not relevant outside this function?

Also, please check the style guides again! [hasOwnProperty](https://github.com/airbnb/javascript#objects--prototype-builtins)